src/averageAges.js

```diff
@@ -37,7 +45,24 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const countOfWomen = women.filter(isMother =>
+    withChildren
+      ? people.some(person => person.mother === isMother.name)
+      : people.some(person => isMother.name === person.name)
+  )
+    .reduce(count => count + 1, 0);
+
+  return women.filter(isMother => {
+    if (withChildren) {
+      return people.some(person => person.mother === isMother.name);
+    } else {
+      return people.some(person => isMother.name === person.name);
+    }
+  })
+    .map(isMother => (isMother.died - isMother.born))
+    .reduce((ageA, ageB) => (ageA + ageB)) / countOfWomen;
```

Comment:
Use the countAverageAge function you created earlier to calculate the average age, you have the same logic used here